code/tests/unittests/test_vector3.cpp

```diff
@@ -156,3 +144,19 @@ TEST_F(Vector3Test, LengthSquaredWhenNoParamsReturnsLengthSquaredValueOfTheVecto
 
     ASSERT_THAT(actualValue, expectedValue);
 }
+
+TEST_P(Vector3ParameterizedTest, BracketOperatorWhenPassingAnIndexReturnsDoubleInThatPosition) {
+    double actual_value= test_vector[GetParam()];
+
+    ASSERT_EQ(actual_value, 1.0);
+}
+
+TEST_P(Vector3ParameterizedTest, BracketOperatorWhenPassingAnIndexReturnsDoubleReferenceInThatPosition) {
+    double* expected_value= std::addressof(test_vector.e[GetParam()]);
+
+    double* actual_value= std::addressof(test_vector[GetParam()]);
+
+    ASSERT_EQ(actual_value, expected_value);
+}
+
+INSTANTIATE_TEST_SUITE_P(BrackerOperators, Vector3ParameterizedTest, ::testing::Values(0, 1, 2));
```

Comment:
I've also changed the name of the test, from `BrackerOperators` to `Vector3BracketOperators`, just because that way it should be easier to identify what the test is about.

Reply:
I agree with the approach, I only modified the test_vector_456 variable to param_test_vector.